pkg/resources/pool.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2018 Intel Corp. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/intel/sriov-network-device-plugin/pkg/types"
+	"github.com/intel/sriov-network-device-plugin/pkg/utils"
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1beta1"
+)
+
+const (
```

Comment:
We can only get sriov PF under /sys/class/net if the PF is bind with supported kernel network driver. For example if a PF is bound with igb_uio driver that will no longer be found in /sys/class/net because this driver does not instantiate a network device. but you can count on /sys/bus/pci/devices as we are now looking at the PCI bus directly. So, the /sys/bus/pci/devices will give you above information(not 'operstate' though) regardless of the driver attached to the device.
We have some usecases where we may want to bind the PF with DPDK supported driver.

Good catch on the constants, I'm going to do some refactoring here.